WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Bases;
+using WalletWasabi.Helpers;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Services
+{
+	public class SystemAwakeChecker : PeriodicRunner
+	{
+		public SystemAwakeChecker(TimeSpan period, WalletManager walletManager) : base(period)
+		{
+			WalletManager = walletManager;
+		}
+
+		private WalletManager WalletManager { get; }
+
+		protected override Task ActionAsync(CancellationToken cancel)
+		{
+			if (WalletManager.IsAnyCoinJoinInProgress())
+			{
+				EnvironmentHelpers.KeepSystemAwake();
+			}
```

Comment:
When coinjoin is not in progress, should this method revert to original OS behavior?

Reply:
There are two ways to use this on windows. The one is that I am using it right now to reset the sleep timer again and again. The other is to set an OS-level flag that will not allow the PC to sleep. To avoid accidental disable of sleep functionality I am using the first approach.

Reply:
At this point, the method should probably be called `ProlongSystemAwakeTimeout()` or something.

Reply:
ACK